hatchways/team-sparta
# Built ui for uploading file

styled front end and prepared to build logic to tie upload to back end

### What this PR does (required):
Created Front end for file upload
Integrated with back end routes so you can push images to AWS S3 Bucket 'teamspartabucket'
Each image has a unique key and aws url that we can use/pass to the db to use throughout our app

### Screenshots / Videos (required):

https://user-images.githubusercontent.com/69362301/121106039-37047d00-c7d3-11eb-8d83-e23e00989645.mp4




![Screen Shot 2021-06-06 at 10 00 15 PM](https://user-images.githubusercontent.com/69362301/120950997-e4b15680-c715-11eb-9a91-a22cb9b9e2c1.png)
![Screen Shot 2021-06-06 at 10 21 59 PM](https://user-images.githubusercontent.com/69362301/120951074-0f9baa80-c716-11eb-8838-fcc64527f482.png)




### Any information needed to test this feature (required):
right now you need to alter one of the routes that are protected to check the component out. Do the following:
1.Navigate to client/src/pages/Dashboard/Dashboard.tsx
2. On line 48, change the rendered component from {ContestForm} to {Submission}
3. You can then use the "Create Contest" button and route to review the component
4. Click the cloud icon and add single or multiple files.
5. Click submit and it'll post to AWS.
6. You can check the console to see the key and url for each image.
7. You can also log into the AWS Bucket to view the same
![Screen Shot 2021-06-06 at 10 26 30 PM](https://user-images.githubusercontent.com/69362301/120951212-538eaf80-c716-11eb-9392-16baa615c92f.png)


### Any issues with the current functionality (optional):
-We'll need to set up and finalize a protected route for this component.
-I couldn't find the exact icon in the mock, went with the closest i could find in the material ui icons library
